### PR TITLE
Change end_of_early_data into a handshake message. This makes all

### PR DESCRIPTION
**Pull request by ekr**
key changes happen at Handshake boundaries.

Suggested by David Benjamin

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -2468,9 +2473,9 @@ MUST be the first PSK listed in the client's "pre_shared_key" extension.
 as their corresponding messages sent in other flights (handshake,
 application_data, and alert respectively) but are protected under
 different keys.  After receiving the server's Finished message, if the
-server has accepted early data, an "end_of_early_data" alert of type
-"warning" MUST be sent to indicate the key change. This message will
-be encrypted with the 0-RTT traffic keys.
+server has accepted early data, an EndOfEarlyData message
+will be sent to indicate key change. This message will be encrypted
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: missing article before "key change"

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -1374,7 +1376,9 @@ processed and transmitted as specified by the current active session state.
            hello_request_RESERVED(0),
            client_hello(1),
            server_hello(2),
+           hello_retry_request_RESERVED(3),
```
Comment (davidben) [inline draft-ietf-tls-tls13.md]: hello_verify_request_RESERVED?

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -3650,15 +3667,6 @@ close_notify
   any more messages on this connection. Any data received after a
   closure MUST be ignored.
 
-end_of_early_data
```
Comment (davidben) [inline draft-ietf-tls-tls13.md]: [This bit was even kind of a lie before. end_of_early_data did not signal "that the connection is ending".]

File: draft-ietf-tls-tls13.md (hunk 2 above)
Comment (ekr) [inline draft-ietf-tls-tls13.md]: Gah, yesh

**Pull request merged**